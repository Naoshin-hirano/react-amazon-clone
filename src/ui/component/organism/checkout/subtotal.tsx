import React from "react";
import "../../../../design/component/atoms/checkout/subtotal.css";
import CurrencyFormat from "react-currency-format";

export const Subtotal: React.FC<any> = ({ basket }) => {
    return (
        <div className="subtotal">
            <CurrencyFormat
                renderText={(value) => (
                    <>
                        <p>
                            Subtotal ({basket?.length} items):{" "}
                            <strong>{value}</strong>
                        </p>
                        <small className="subtotal__gift">
                            <input type="checkbox" /> This order contains a gift
                        </small>
                    </>
                )}
                decimalScale={2}
                value="76,000"
                displayType={"text"}
                thousandSeparator={true}
                prefix={"$"}
            />
            {/* 支払い画面へ遷移 */}
            <button>Proceed to Checkout</button>
        </div>
    );
};
