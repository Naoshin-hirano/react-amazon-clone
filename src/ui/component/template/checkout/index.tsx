import React from "react";
import { Checkout } from "ui/component/organism/checkout/checkout";

export const Template: React.FC<any> = (props) => {
    return (
        <div>
            <Checkout {...props} />
        </div>
    );
};
