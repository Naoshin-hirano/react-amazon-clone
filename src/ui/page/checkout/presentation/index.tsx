export const useCreateProps = () => {
    return {
        secondTitle,
        secondMenu,
        backButton,
        basket,
    };
};

type SecondTitle = {
    title: string;
};
type SecondMenu = {
    buttonLabels: {
        menu1: string;
        menu2: string;
        menu3: string;
    };
};
type BackButton = {
    buttonLabels: {
        back: string;
    };
};
type Basket = {
    id: string;
    title: string;
    image: string;
    price: string;
    rating: number;
}[];

const secondTitle: SecondTitle = {
    title: "セカンドタイトル",
};
const secondMenu: SecondMenu = {
    buttonLabels: {
        menu1: "セカンドメニュー1",
        menu2: "セカンドメニュー2",
        menu3: "セカンドメニュー3",
    },
};
const backButton: BackButton = {
    buttonLabels: {
        back: "閉じる",
    },
};

const basket: Basket = [
    {
        id: "fgkammrngoalenaeg",
        title: "登山グッズ一式",
        image:
            "https://www.anoyama.com/wp-content/uploads/2019/05/0_equipment-860x550.jpg",
        price: "2,340",
        rating: 4,
    },
    {
        id: "fgkameggaeregaeg",
        title: "キャンプグッズ一式",
        image:
            "https://images.camphack.jp/wp-content/uploads/2022/02/image19.jpeg",
        price: "5,300",
        rating: 5,
    },
    {
        id: "thhjskmeggaeregaeg",
        title: "登山シューズ",
        image: "https://m.media-amazon.com/images/I/616AAPLKJTL._AC_UY500_.jpg",
        price: "16,400",
        rating: 3,
    },
];
