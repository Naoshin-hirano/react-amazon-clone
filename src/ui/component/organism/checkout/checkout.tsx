import React from "react";
import "../../../../design/component/atoms/checkout/checkout.css";
import { Subtotal } from "./subtotal";
import { CheckoutProduct } from "./checkout-product";

export const Checkout: React.FC<any> = (props) => {
    const { basket } = props;
    // pagesのstoreからuseSelectorで取得したデータをpropsで取得
    // const [{ basket, user }, dispatch] = useStateValue();
    return (
        <div className="checkout">
            <div className="checkout__left">
                <img
                    className="checkout__ad"
                    src="https://images-na.ssl-images-amazon.com/images/G/02/UK_CCMP/TM/OCC_Amazon1._CB423492668_.jpg"
                    alt="checkout component ad banner"
                />

                <div>
                    <h3>Hello xxx@yyy.com</h3>
                    <h2 className="checkout__title">Your shopping basket</h2>

                    {basket.map((item: any, key: any) => (
                        <CheckoutProduct
                            key={key}
                            id={item.id}
                            title={item.title}
                            image={item.image}
                            price={item.price}
                            rating={item.rating}
                        />
                    ))}
                </div>
            </div>

            <div className="checkout__right">
                <Subtotal backet={basket} />
            </div>
        </div>
    );
};
