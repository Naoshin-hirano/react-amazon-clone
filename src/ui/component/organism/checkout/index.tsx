import React from "react";
import "../../../../design/component/atoms/checkout/index.css";
import { Subtotal } from "./subtotal";
import { CheckoutProduct } from "../common/checkout-product";
import { CartProduct } from "core/store/ui/top/type";

export const Checkout: React.FC<any> = (props) => {
    const { busketInfo } = props;
    return (
        <div className="checkout">
            <div className="checkout__left">
                <img
                    className="checkout__ad"
                    src="https://images-na.ssl-images-amazon.com/images/G/02/UK_CCMP/TM/OCC_Amazon1._CB423492668_.jpg"
                    alt="checkout component ad banner"
                />

                <div>
                    <h3>平野直弥さん</h3>
                    <h2 className="checkout__title">カート内の商品</h2>

                    {busketInfo.productInCart ? (
                        busketInfo.productInCart.map(
                            (item: CartProduct, key: string) => (
                                <CheckoutProduct
                                    key={key}
                                    id={item.id}
                                    title={item.title}
                                    image={item.image}
                                    price={item.price}
                                    rating={item.rating}
                                />
                            )
                        )
                    ) : (
                        <>商品が存在しません</>
                    )}
                </div>
            </div>

            <div className="checkout__right">
                <Subtotal busketInfo={busketInfo} />
            </div>
        </div>
    );
};
