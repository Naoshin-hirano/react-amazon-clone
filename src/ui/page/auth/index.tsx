import React from "react";
import { Template } from "ui/component/template/auth";
import { useCreateProps } from "./presentation";

export const AuthPage: React.FC = () => {
    const props = useCreateProps();
    return <Template {...props}></Template>;
};
