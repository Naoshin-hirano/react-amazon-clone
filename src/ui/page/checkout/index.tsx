import React from "react";
import { Template } from "ui/component/template/checkout";
import { useCreateProps } from "./presentation";

export const CheckoutPage: React.FC = () => {
    const props = useCreateProps();
    return <Template {...props}></Template>;
};
