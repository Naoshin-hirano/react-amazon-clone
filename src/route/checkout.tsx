import { Route } from "@tanstack/react-location";
import React from "react";
// import { SocondSamplePage } from "ui/page/second";
import { CheckoutPage } from "ui/page/checkout";

export const checkoutRoute: Route[] = [
    {
        path: "/checkout",
        element: <CheckoutPage />,
    },
    // {
    //     path: "/resigter",
    //     element: <SocondSamplePage />,
    // },
];
