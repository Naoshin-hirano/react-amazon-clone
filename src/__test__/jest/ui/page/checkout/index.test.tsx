import { render } from "@testing-library/react";
import React from "react";
import { CheckoutPage } from "ui/page/checkout";

describe("Checkout", () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it("Checkout画面 snapshot確認", async () => {
        const { asFragment } = render(<CheckoutPage />);
        expect(asFragment()).toMatchSnapshot("Checkout");
    });
});
