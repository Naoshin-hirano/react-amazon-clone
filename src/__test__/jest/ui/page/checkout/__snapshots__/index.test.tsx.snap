// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Checkout画面 snapshot確認: Checkout 1`] = `
<DocumentFragment>
  <div>
    <div
      class="checkout"
    >
      <div
        class="checkout__left"
      >
        <img
          alt="checkout component ad banner"
          class="checkout__ad"
          src="https://images-na.ssl-images-amazon.com/images/G/02/UK_CCMP/TM/OCC_Amazon1._CB423492668_.jpg"
        />
        <div>
          <h3>
            Hello xxx@yyy.com
          </h3>
          <h2
            class="checkout__title"
          >
            Your shopping basket
          </h2>
          <div
            class="checkoutProduct"
          >
            <img
              alt="checkoutProductImg"
              class="checkoutProduct__image"
              src="https://www.anoyama.com/wp-content/uploads/2019/05/0_equipment-860x550.jpg"
            />
            <div
              class="checkoutProduct__info"
            >
              <p
                class="checkoutProduct__title"
              >
                登山グッズ一式
              </p>
              <p
                class="checkoutProduct__price"
              >
                <small>
                  $
                </small>
                <strong>
                  23,400
                </strong>
              </p>
              <div
                class="checkoutProduct__rating"
              >
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
              </div>
              <button>
                Remove from Basket
              </button>
            </div>
          </div>
          <div
            class="checkoutProduct"
          >
            <img
              alt="checkoutProductImg"
              class="checkoutProduct__image"
              src="https://images.camphack.jp/wp-content/uploads/2022/02/image19.jpeg"
            />
            <div
              class="checkoutProduct__info"
            >
              <p
                class="checkoutProduct__title"
              >
                キャンプグッズ一式
              </p>
              <p
                class="checkoutProduct__price"
              >
                <small>
                  $
                </small>
                <strong>
                  53,400
                </strong>
              </p>
              <div
                class="checkoutProduct__rating"
              >
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
              </div>
              <button>
                Remove from Basket
              </button>
            </div>
          </div>
          <div
            class="checkoutProduct"
          >
            <img
              alt="checkoutProductImg"
              class="checkoutProduct__image"
              src="https://m.media-amazon.com/images/I/616AAPLKJTL._AC_UY500_.jpg"
            />
            <div
              class="checkoutProduct__info"
            >
              <p
                class="checkoutProduct__title"
              >
                登山シューズ
              </p>
              <p
                class="checkoutProduct__price"
              >
                <small>
                  $
                </small>
                <strong>
                  16,400
                </strong>
              </p>
              <div
                class="checkoutProduct__rating"
              >
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
                <p>
                  🌟
                </p>
              </div>
              <button>
                Remove from Basket
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="checkout__right"
      >
        <div
          class="subtotal"
        >
          <p>
            Subtotal ( items): 
            <strong>
              $76,000
            </strong>
          </p>
          <small
            class="subtotal__gift"
          >
            <input
              type="checkbox"
            />
             This order contains a gift
          </small>
          <button>
            Proceed to Checkout
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
